tst/regression/CMakeLists.txt

```diff
@@ -49,6 +52,23 @@ list(APPEND TEST_DIRS calculate_pi)
 list(APPEND TEST_ARGS "--driver ${CMAKE_BINARY_DIR}/example/calculate_pi/pi-example \
 --driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/calculate_pi/parthinput.regression")
 
+# Advection test (here added manually to override default nproc for MPI runs)
+setup_test( advection_convergence "--driver ${CMAKE_BINARY_DIR}/example/advection/advection-example \
+--driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/advection_convergence/parthinput.advection \
+--num_steps 24")
+setup_test_mpi( 1 advection_convergence "--driver ${CMAKE_BINARY_DIR}/example/advection/advection-example \
+--driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/advection_convergence/parthinput.advection \
+--num_steps 24")
+
+if (ENABLE_HDF5)
+setup_test( output_hdf5 "--driver ${CMAKE_BINARY_DIR}/example/advection/advection-example \
+--driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/output_hdf5/parthinput.advection \
+--num_steps 2")
+setup_test_mpi( ${NUM_MPI_PROC_TESTING} output_hdf5
+"--driver ${CMAKE_BINARY_DIR}/example/advection/advection-example \
+--driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/output_hdf5/parthinput.advection \
+--num_steps 2")
+endif()
```

Comment:
I think this would be a lot cleaner, if you just added a separate list for MPI processors and used the existing for loop below. But it would also be nice to know why you need to hard code a single mpi processor for the advection test? The variable NUM_MPI_PROC_TESTING can always be changed when building.

Reply:
I had to hard code using a single proc here as the tests cover the case that there's only one meshblock covering the entire mesh. Thus, using more than 1 process will fail. It's an edge case, but still worth including in the test from my point of view.